pyfarm/master/api/jobgroups.py

```diff
@@ -0,0 +1,355 @@
+        """
+        out = []
+        for jobgroup in JobGroup.query:
+            jobgroup_data = jobgroup.to_dict(
+                unpack_relationships=["user", "main_jobtype"])
+            del jobgroup_data["user_id"]
+            del jobgroup_data["main_jobtype_id"]
+            out.append(jobgroup_data)
+
+        return jsonify(out), OK
+
+
+class SingleJobGroupAPI(MethodView):
+    def get(self, group_id):
+        """
+        A ``GET`` to this endpoint will return the requested job group
+
+        .. http:get:: /api/v1/jobgroups/<int:id> HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                GET /api/v1/jobgroups/2 HTTP/1.1
+                Accept: application/json
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 200 OK
+                Content-Type: application/json
+
+                {
+                    "id": 2,
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType",
+                    "jobs": [],
+                    "title": "Test Group"
+                }
+
+        :statuscode 200: no error
+        :statuscode 404: the requested job group was not found
+        """
+        jobgroup = JobGroup.query.filter_by(id=group_id).first()
+
+        if not jobgroup:
+            return (jsonify(error="Requested job group %s not found" % group_id),
+                    NOT_FOUND)
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
+        del jobgroup_data["main_jobtype_id"]
+
+        return jsonify(jobgroup_data), OK
+
+    def post(self, group_id):
+        """
+        A ``POST`` to this endpoint will update the specified group with the data
+        in the request.  Columns not specified in the request will be left as
+        they are.
+
+        .. http:post:: /api/v1/jobgroups/<int:id> HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                POST /api/v1/jobgroups/2 HTTP/1.1
+                Accept: application/json
+
+                {
+                    "user": "testuser2"
+                }
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 201 OK
+                Content-Type: application/json
+
+                {
+                    "id": 2,
+                    "user": "testuser2",
+                    "main_jobtype": "Test JobType",
+                    "jobs": [],
+                    "title": "Test Group"
+                }
+
+        :statuscode 200: the job group was updated
+        :statuscode 400: there was something wrong with the request (such as
+                         invalid columns being included)
+        """
+        jobgroup = JobGroup.query.filter_by(id=group_id).first()
+
+        if not jobgroup:
+            return (jsonify(error="Requested job group %s not found" % group_id),
+                    NOT_FOUND)
+
+        if "title" in g.json:
```

Comment:
Not a good idea. This will set ```title``` to None if it's not present in the request, but since this is an update, the correct way to handle that is to just leave ```title``` as it is.